src/common/str_map.cc

```diff
@@ -57,6 +57,9 @@ int get_json_str_map(
   return 0;
 }
 string trim(const string& str) {
+  if (str.size()) {
```

Comment:
Serves me right for making assumptions.

Reply:
or we can just put 
```c++
  return boost::algorithm::trim_copy_if(str,
                                        [](unsigned char c) {
                                          return std::isspace(c);
                                        });
```

without involving `std::locale`.